homeassistant/components/idasen_desk/__init__.py

```diff
@@ -0,0 +1,84 @@
+"""The IKEA Idasen Desk integration."""
+from __future__ import annotations
+
+import logging
+
+from bleak import BleakError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_NAME,
+    CONF_ADDRESS,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+from .desk import Desk, DeskData
+
+PLATFORMS: list[Platform] = [Platform.COVER]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up IKEA Idasen from a config entry."""
+    address: str = entry.data[CONF_ADDRESS]
+
+    coordinator: DataUpdateCoordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=entry.title,
+    )
+
+    def handle_desk_update(height_percent: int | None) -> None:
+        coordinator.async_set_updated_data(height_percent)
+
+    desk = Desk(hass, address.upper(), handle_desk_update)
```

Comment:
```suggestion
    desk = Desk(hass, address.upper(), coordinator.async_set_updated_data)
```

Do you need the extra function?

Reply:
Not needed indeed